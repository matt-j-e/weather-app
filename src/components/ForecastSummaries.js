import React from "react";
import PropTypes from "prop-types";
import "../styles/ForecastSummaries.css";
import ForecastSummary from "./ForecastSummary";

const ForecastSummaries = ({ forecasts }) => {
  return (
    <div className="forecast-summaries">
      {forecasts.map((forecast) => {
        return <ForecastSummary forecast={forecast} key={forecast.date} />;
      })}
    </div>
  );
};

ForecastSummaries.propTypes = {
  forecasts: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default ForecastSummaries;
