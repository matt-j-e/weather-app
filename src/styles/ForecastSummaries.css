.forecast-summaries {
    /* background-color: yellowgreen; */
    max-width: 100vw;
    overflow: hidden;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}

/*** MEDIA QUERIES ***/
@media screen and (min-width: 650px) {
    .forecast-summaries {
        margin-top: 4rem;
    }
}