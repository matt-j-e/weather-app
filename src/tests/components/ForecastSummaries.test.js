import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = {
    forecasts: [
      {
        date: 111,
        description: "Stub description 1",
        icon: "800",
        temperature: {
          max: 22,
          min: 11,
        },
      },
      {
        date: 222,
        description: "Stub description 2",
        icon: "211",
        temperature: {
          max: 24,
          min: 9,
        },
      },
    ],
    onForecastSelect: jest.fn(),
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummaries
        forecasts={validProps.forecasts}
        onForecastSelect={validProps.onForecastSelect}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct number of tests", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries
        forecasts={validProps.forecasts}
        onForecastSelect={validProps.onForecastSelect}
      />
    );
    expect(getAllByTestId("forecast-summary").length).toBe(2);
  });
});
