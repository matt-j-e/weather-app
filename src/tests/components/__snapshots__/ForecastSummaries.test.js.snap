// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
      role="button"
      tabindex="0"
    >
      <div
        class="forecast-summary__date"
      >
        Thu
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-sunny"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        22°c
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description 1
      </div>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
      role="button"
      tabindex="0"
    >
      <div
        class="forecast-summary__date"
      >
        Thu
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-thunderstorm"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        24°c
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description 2
      </div>
    </div>
  </div>
</DocumentFragment>
`;
